frontend/src/lib/components/Alerts/views/EditAlertModal.tsx

```diff
@@ -0,0 +1,256 @@
+                        <th className="text-right">Time</th>
+                        <th className="text-right pr-4">Value</th>
+                        <th>Targets notified</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {alert.checks.map((check) => (
+                        <tr key={check.id}>
+                            <td>{check.state}</td>
+                            <td className="text-right">
+                                <TZLabel time={check.created_at} />
+                            </td>
+                            <td className="text-right pr-4">{check.calculated_value}</td>
+                            <td>{check.targets_notified ? 'Yes' : 'No'}</td>
+                        </tr>
+                    ))}
+                </tbody>
+            </table>
+        </div>
+    )
+}
+
+export function EditAlertModal({ alertId, onClose, isOpen, onEditSuccess, insight }: EditAlertProps): JSX.Element {
+    if (!alertId && !insight) {
+        throw new Error('Must provide either an alertId or an insight to create alert for')
+    }
+
+    const { alert, alertLoading } = useValues(alertLogic({ alertId }))
+
+    if (alertLoading) {
+        return <SpinnerOverlay />
+    }
+
+    return (
+        <LemonModal onClose={onClose} isOpen={isOpen} width={600} simple title="">
+            {alert ? (
+                <EditAlertForm
+                    alert={alert}
+                    onClose={onClose}
+                    onEditSuccess={onEditSuccess}
+                    alertState={<AlertStateTable alert={alert} />}
+                />
+            ) : alertId !== undefined ? (
+                // we have alertId but didn't get alert
+                <div className="p-4 text-center">
+                    <h2>Not found</h2>
+                    <p>This alert could not be found. It may have been deleted.</p>
+                </div>
+            ) : // we need to create a new alert
+            insight === undefined ? (
+                <div className="p-4 text-center">
+                    <p>Can only create new alert in context of an insight</p>
+                </div>
+            ) : (
+                <EditAlertForm
+                    alert={
+                        {
+                            id: undefined,
+                            name: '',
+                            created_by: null,
+                            created_at: '',
+                            enabled: true,
+                            threshold: {
+                                configuration: {
+                                    absoluteThreshold: {},
+                                },
+                            },
+                            subscribed_users: [],
+                            checks: [],
+                            insight: insight.id,
+                            insight_short_id: insight.short_id,
+                        } as AlertFormType
```

Comment:
It's probably better to pass insight and an optional alert. Best practice is to avoid creating objects in props, but it's not very important right now because we don't really optimize frontend😁

Reply:
Thanks! This is very helpful. Was wondering what's the best way to structure the logic around kea-forms. Would you then suggest to create the default `AlertFormType` [here](https://github.com/PostHog/posthog/pull/25056/files#diff-f09024b6e40090341024e27c19ee32f7c4b0e8af799075f45497a5317c75e250R44)?

Or should I rewrite the `alertFormLogic` to something better?

Reply:
Yeah, I would assign a default form's object in the `alertFormLogic` and pass from props required variables (id and short_id).

Reply:
Thanks for all the help here, been instrumental in me cleaning up the form. The final stage is just passing the alert id in the form but weirdly kea runs into weird errors trying to use a loader or a connect logic in the form (only the scenario on the insight page, (alerts tab works fine)). Spent a few hours on it, so will save that enhancement for a later PR so I can test this with users first :)